src/main/java/com/osiris/autoplug/client/tasks/updater/plugins/TaskPluginDownload.java

```diff
@@ -32,32 +32,33 @@
 import java.util.Arrays;
 
 
-public class TaskPluginDownload extends BThread {
+public class TaskPluginDownload extends TaskDownloadBase {
```

Comment:
same here, is it the correct class to extend from?

Reply:
yeah thats great, however I think the duplicate code still exists, meaning it was not removed from the extending classes and put into TaskDownload.
I think it wasn't as easy, thats why originally I was extending BThread and not TaskDownload directly. Let me know if thats not the case and if you can do that.